acceptance/tests/options/config_file/blocklist.rb

```diff
@@ -0,0 +1,57 @@
+# This test verifies that when a fact group is blocked in the config file
+# the corresponding facts do not resolve.
+test_name "facts can be blocked via a list in the config file" do
+  require 'facter/acceptance/user_fact_utils'
+  extend Facter::Acceptance::UserFactUtils
+
+  agents.each do |agent|
+    step "facts should be blocked when Facter is run from the command line" do
+      step "Agent #{agent}: create config file" do
+        custom_conf_dir = agent.tmpdir("config_dir")
+        config_file = File.join(custom_conf_dir, "facter.conf")
+        create_remote_file(agent, config_file, <<-FILE)
+        facts : { blocklist : [ "file system" ] }
+        FILE
+
+        teardown do
+          on(agent, "rm -rf '#{custom_conf_dir}'", :acceptable_exit_codes => [0, 1])
+        end
+
+        step "blocked facts should not be resolved" do
```

Comment:
Would like to add an additional test for debug output shows when a fact is being blocked e.g.

Add `cli: { debug : true }` to config file

```
assert_match(/blocking collection of .+ facts/, stderr, "Expected stderr to contain statement about blocking fact collection")
```